pycbc/events/significance.py

```diff
@@ -187,6 +187,19 @@ def get_n_louder(back_stat, fore_stat, dec_facs,
     Wrapper to find the correct n_louder calculation method using standard
     inputs
     """
+    # This nan to inf conversion is a safety shield; there shouldn't be any
+    # nan statistic values, but if there are, they should be considered as
+    # quiet as possible. nans are considered greater than any floats in an
+    # argsort
+    nanmask = np.isnan(back_stat)
+    if any(nanmask):
+        logging.warning(
+            "Removing %d NaN background statistic values",
+            np.count_nonzero(nanmask),
+        )
+        back_stat = copy.deepcopy(back_stat)
```

Comment:
I don't think this line does anything .. ?? 
At the end of the day I don't think there is any reason to not modify back_stat in place either

Reply:
the deepcopy will mean that the local variable points to a different memory than the input variable, so changes to it will not affect the input variable.

My reasoning for not modifying back_stat in-place is that I would not expect a function called `get_n_louder` to modify the background statistic in place, and I don't think it is sensible to do that

Reply:
mmm .. regardless of whether it does what we want wrt modifying variables or not, `back_stat = copy.deepcopy(back_stat)` is deeply confusing code IMO.